src/ios/SentryCordova.m

```diff
@@ -8,121 +8,245 @@
+
+    NSError *error = nil;
+
+    SentryOptions *sentryOptions = [[SentryOptions alloc] initWithDict:options
+                                                      didFailWithError:&error];
+
+    CDVPluginResult *result =
+        [CDVPluginResult resultWithStatus:CDVCommandStatus_OK
+                            messageAsBool:YES];
+    if (error != nil) {
+      NSLog(@"%@", error);
+      result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR
+                                   messageAsBool:NO];
+    } else {
+      [SentrySDK startWithOptionsObject:sentryOptions];
     }
-    event.sdk = @{@"name": SentryCordovaSdkName,
-                  @"version": SentryCordovaVersionString,
-                  @"integrations": @[@"sentry-cocoa"]};
+
+    [self.commandDelegate sendPluginResult:result
+                                callbackId:command.callbackId];
+  }];
 }
 
-- (void)sendEvent:(CDVInvokedUrlCommand *)command {
-    [self.commandDelegate runInBackground:^{
-        NSDictionary *jsonEvent = [command.arguments objectAtIndex:0];
-        if ([NSJSONSerialization isValidJSONObject:jsonEvent]) {
-            NSData *jsonData = [NSJSONSerialization dataWithJSONObject:jsonEvent
-                                                               options:0
-                                                                 error:nil];
-
-            SentryEvent *sentryEvent = [[SentryEvent alloc] initWithJSON:jsonData];
-            if (SentryClient.sharedClient != nil) {
-                [SentryClient.sharedClient sendEvent:sentryEvent withCompletionHandler:^(NSError * _Nullable error) {
-                    CDVPluginResult *result = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsBool:YES];
-                    if (error != nil) {
-                        result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsBool:NO];
-                    }
-                    [self.commandDelegate sendPluginResult:result callbackId:command.callbackId];
-                }];
-            }
+- (void)setReleaseVersionDist:(SentryEvent *)event {
+  if (event.extra[@"__sentry_version"]) {
+    NSDictionary *infoDict = [[NSBundle mainBundle] infoDictionary];
+    event.releaseName =
+        [NSString stringWithFormat:@"%@-%@", infoDict[@"CFBundleIdentifier"],
+                                   event.extra[@"__sentry_version"]];
+  }
+  if (event.extra[@"__sentry_release"]) {
+    event.releaseName =
+        [NSString stringWithFormat:@"%@", event.extra[@"__sentry_release"]];
+  }
+  if (event.extra[@"__sentry_dist"]) {
+    event.dist =
+        [NSString stringWithFormat:@"%@", event.extra[@"__sentry_dist"]];
+  }
+  event.sdk = @{
+    @"name" : SentryCordovaSdkName,
+    @"version" : SentryCordovaVersionString,
+    @"integrations" : @[ @"sentry-cocoa" ]
+  };
+}
+
+- (void)captureEnvelope:(CDVInvokedUrlCommand *)command {
+  [self.commandDelegate runInBackground:^{
```

Comment:
Is the unhandled error also going through here? I wonder if the app can crash before the SDK gets a hold of it since it seems to be queued to a background task

Reply:
Yes, it would be going through here as well. This was adapted from the previous code, should this be in the UI thread then?